extensions/mssql/src/utils.ts

```diff
@@ -348,3 +356,73 @@ export async function getOrDownloadServer(config: IConfig, handleServerEvent?: (
 
 	return serverdownloader.getOrDownloadServer();
 }
+
+/**
+ * Returns the credential folder path on the user's system
+ */
+export async function removeCredentialFile(): Promise<void> {
+	const home = os.homedir();
+	const credentialPath = path.join(home, '.sqlsecrets');
+	await fs.rmdir(credentialPath, { recursive: true });
+}
+
+/**
+ *
+ * @returns Whether the current OS is linux or not
+ */
+export function isLinux(): boolean {
+	return os.platform() === 'linux';
+}
+
+function getConfigUseNativeCredentials(): boolean {
+	let config = getConfiguration();
+	if (config) {
+		return Boolean(config[configUseNativeCredentials]);
+	} else {
+		return undefined;
+	}
+}
+
+/**
+ * This function returns whether the native credential management system
+ * should be used instead of tools service
+ */
+export function useNativeCredentialsEnabled(): boolean {
+	const linux: boolean = isLinux();
```

Comment:
Why only for Linux? It would generally be easier to work with if we're storing and retrieving secrets in the same way for the app no matter the platform.